config/confighttp/confighttp.go

```diff
@@ -271,6 +272,10 @@ type HTTPServerConfig struct {
 	// Additional headers attached to each HTTP response sent to the client.
 	// Header values are opaque since they may be sensitive.
 	ResponseHeaders map[string]configopaque.String `mapstructure:"response_headers"`
+
+	// MemoryLimiter is memory limiter this receiver will use to restrict incoming requests
+	// currently in development and does nothing
+	MemoryLimiter *memorylimiterextension.MemoryLimiter `mapstructure:"mem_limiter"`
```

Comment:
Let's use `*component.ID` here. See https://github.com/open-telemetry/opentelemetry-collector/blob/5bcd109c718fc9f11f070ae66c79770629854872/exporter/exporterhelper/queue_sender.go#L39 for reference

Reply:
updated